### PR TITLE
fix: redirect to 404 page for invalid email verification token

### LINKED ISSUE #9469: Add a page for invalid token while verifying email.
### Is your proposal related to a problem?
While verifying the email on signup, instead of clicking directly on the email link, if they copy the link and accidently add new characters in the token, they see token Not Found error as json response.
Visit /api/auth/verify-email?token=validToken
[Image link](https://imgur.com/a/jV3XkVU)
Refer to feature #9081 
### Describe the solution you'd like

 Instead of showing the json response to the user, we should show them a page that tells the user that the token is invalid/not found.

### Describe alternatives you've considered

Clicking on the link is always works, but some users have a habit not to click on links sent in the emails.

### PR DESCRIPTION
**Pull request by iamr-kumar**
## What does this PR do?

Fixes #9469 

## Type of change

- Bug fix (non-breaking change which fixes an issue)
Screenshot 2023-06-19 at 12 01 55 AM

## How should this be tested?

 
- [x] Send email verification link. Copy paste the link in browser and modify the token.
- [x] Go to `/api/auth/verify-email?token={some-invalid-token}`

## Mandatory Tasks
- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

**Review by Udit-takkar — CHANGES_REQUESTED** [review]
well this is what is see when i enter incorrect token in param. The text says username /404 does not exist 

Screenshot 2023-06-16 at 2 35 37 PM

we should display correct text here like "Your token has expired".
you can check /pages/404 page to see how you can show custom message based on condition

**Comment by iamr-kumar** [comment]
Thanks for the input. Also could you help me with how to get the strings translated? Or is it done by the maintainers?

**Comment by iamr-kumar** [comment]
Update the 404 page

**Comment by Udit-takkar** [comment]
> Thanks for the input. Also could you help me with how to get the strings translated? Or is it done by the maintainers?

You just have to create string in english common/en.json file and use it like `t("page_does_not_exist")`  in the code.

**Comment by iamr-kumar** [comment]
> > Thanks for the input. Also could you help me with how to get the strings translated? Or is it done by the maintainers?
> 
> You just have to create string in english common/en.json file and use it like `t("page_does_not_exist")` in the code.

My point was if the translation to other languages is automated or not. But thanks anyway. I didn't need to create any new string as found some existing ones that fit the use case :)

**Review by Udit-takkar — APPROVED** [review]
LGTM. works fine

**Review by sean-brydon — APPROVED** [review]
thank you! Nice work

File: apps/web/pages/404.tsx [hunk 1]
```diff
@@ -42,6 +42,10 @@ export default function Custom404() {
   const isSubpage = router.asPath.includes("/", 2) || isSuccessPage;
   const isSignup = router.asPath.startsWith("/signup");
   const isCalcom = process.env.NEXT_PUBLIC_WEBAPP_URL === "https://app.cal.com";
+
+  // In case of invalid email verification token, we intentionally redirect to 404 from the API
+  const isInvalidToken = router.asPath === "/404";
```
Comment (zomars) [inline apps/web/pages/404.tsx:47]: @sean-brydon @Udit-takkar this is causing other 404 pages to show as invalid token when it's not the case. Will revert this PR.

**Pull request merged**